app/src/main/java/com/ae/apps/tripmeter/fragments/TripsListFragment.java

```diff
@@ -103,10 +154,30 @@ public void onClick(View v) {
                 showAddTripDialog();
             }
         });
-
         return view;
     }
 
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSIONS_REQUEST_READ_CONTACTS: {
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    //TODO: Reload this Fragment.Otherwise the FAB won't work
+                } else {
+                    Toast.makeText(getActivity(), "This feature needs access to contacts.", Toast.LENGTH_LONG).show();
+                    //TODO: Direct user to price fragment as this fragment cannot be used
```

Comment:
lets create a new TextView saying "Need permission for accessing contacts for this feature to work" and remain in the same view.